app/test_main.py

```diff
@@ -2,0 +1,19 @@
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@mock.patch("app.main.valid_google_url")
+def test_can_access_google_page_with_valid_url(
+        mocked_url: mock.MagicMock
+) -> None:
+    can_access_google_page("https://www.google.com/")
+    mocked_url.assert_called_once_with("https://www.google.com/")
+
+
+@mock.patch("app.main.has_internet_connection")
+def test_can_access_google_page_with_internet_connection(
+        mocked_internet: mock.MagicMock
+) -> None:
```

Comment:
The test name `test_can_access_google_page_with_internet_connection` could be more descriptive. Consider renaming it to something like `test_accessible_with_internet_connection` to better reflect the functionality being tested, as per the checklist's recommendation for descriptive test names.